include/qemu/log.h

```diff
@@ -64,6 +64,7 @@ static inline bool qemu_log_separate(void)
 /* LOG_STRACE is used for user-mode strace logging. */
 #define LOG_STRACE         (1 << 19)
 #define CPU_LOG_INSTR      (1 << 20)
+#define CPU_LOG_INSTR_U    (1 << 21)
```

Comment:
Why this needed to be added among the top-level qemu tracing modes? My rationale for removing it a while back was that we would encode the mode per-cpu in the log state and when requested either by using NOPs or by using the qemu monitor we could call into the `cpu_loglevel_switch` functions.

Reply:
I was not aware do_cpu_loglevel_switch could be accessed from the qemu monitor. I added this pretty much because I wanted a way to enabled only userspace logging from the QEMU console / on startup with a command line argument.

Reply:
Probably need to expose the function as it is currently internal. Leave it an I can change it later. I'm planning some work on the tracing side to support @heshamelmatary analysis anyway.